Sources/SwiftBuildSupport/SwiftBuildSystem.swift

```diff
@@ -174,6 +174,388 @@ private final class PlanningOperationDelegate: SWBPlanningOperationDelegate, Sen
+                throw Diagnostics.fatalError
+            }
+            return target
+        }
+
+        func taskSignature(for id: Int) -> String? {
+            if let signature = taskIDToSignature[id] {
+                return signature
+            }
+            return nil
+        }
+
+        mutating func appendToBuffer(_ task: String, data: Data) {
+            taskBuffer[task, default: .init()].append(data)
+        }
+
+        func dataBuffer(for task: SwiftBuild.SwiftBuildMessage.TaskStartedInfo) -> Data? {
+            guard let data = taskBuffer[task.taskSignature] else {
+                // If there is no available buffer, simply return nil.
+                return nil
+            }
+
+            return data
+        }
+    }
+
+    private func emitInfoAsDiagnostic(info: SwiftBuildMessage.DiagnosticInfo) {
+        // Assure that we haven't already emitted this diagnostic.
+        let fixItsDescription = if info.fixIts.hasContent {
+            ": " + info.fixIts.map { String(describing: $0) }.joined(separator: ", ")
+        } else {
+            ""
+        }
+        let message = if let locationDescription = info.location.userDescription {
+            "\(locationDescription) \(info.message)\(fixItsDescription)"
+        } else {
+            "\(info.message)\(fixItsDescription)"
+        }
+        let severity: Diagnostic.Severity = switch info.kind {
+        case .error: .error
+        case .warning: .warning
+        case .note: .info
+        case .remark: .debug
+        }
+        self.observabilityScope.emit(severity: severity, message: "\(message)\n")
+
+        for childDiagnostic in info.childDiagnostics {
+            emitInfoAsDiagnostic(info: childDiagnostic)
+        }
+    }
+
+    /// Represents a parsed diagnostic segment from compiler output
+    private struct ParsedDiagnostic {
+        /// The file path if present
+        let filePath: String?
+        /// The line number if present
+        let line: Int?
+        /// The column number if present
+        let column: Int?
+        /// The severity (error, warning, note, remark)
+        let severity: String
+        /// The diagnostic message text
+        let message: String
+        /// The full text including any multi-line context (code snippets, carets, etc.)
+        let fullText: String
+
+        /// Parse severity string to Diagnostic.Severity
+        func toDiagnosticSeverity() -> Basics.Diagnostic.Severity {
+            switch severity.lowercased() {
+            case "error": return .error
+            case "warning": return .warning
+            case "note": return .info
+            case "remark": return .debug
+            default: return .info
+            }
+        }
+    }
+
+    /// Split compiler output into individual diagnostic segments
+    /// Format: /path/to/file.swift:line:column: severity: message
+    private func splitIntoDiagnostics(_ output: String) -> [ParsedDiagnostic] {
```

Comment:
I don't think we should be doing _any_ output parsing at this level, Swift Build already has an entire subsystem dedicated to doing this.